engine/consensus/dkg/reactor_engine.go

```diff
@@ -423,7 +423,7 @@ func (e *ReactorEngine) end(nextEpochCounter uint64) func() error {
 		if crypto.IsDKGFailureError(err) {
 			e.log.Warn().Err(err).Msgf("node %s with index %d failed DKG locally", e.me.NodeID(), e.controller.GetIndex())
 			err := e.dkgState.SetDKGEndState(nextEpochCounter, flow.DKGEndStateDKGFailure)
-			if err != nil {
+			if err != nil && !errors.Is(err, storage.ErrAlreadyExists) {
```

Comment:
added just in case handling of sentinel error if recovery happened before failing the DKG